src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/TelekineticUtilitySpell.java

```diff
@@ -0,0 +1,71 @@
+package com.mineboundteam.minebound.magic.UtilitySpells;
+
+import com.mineboundteam.minebound.MineBound;
+import com.mineboundteam.minebound.config.IConfig;
+import com.mineboundteam.minebound.item.armor.MyrialArmorItem;
+import com.mineboundteam.minebound.magic.PassiveSpellItem;
+import com.mineboundteam.minebound.magic.SpellLevel;
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.common.ForgeConfigSpec.Builder;
+import net.minecraftforge.common.ForgeConfigSpec.IntValue;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class TelekineticUtilitySpell extends PassiveSpellItem {
+
+    private final int manaCost;
+    private final SpellLevel level;
+
+    public TelekineticUtilitySpell(Properties properties, TelekineticUtilitySpellConfig config) {
+        super(properties);
+
+        manaCost = config.MANA_COST.get();
+        level = config.LEVEL;
+    }
+
+    @Override
+    public void updateEquipment(MyrialArmorItem item, Player player) {
+        super.updateEquipment(item, player);
+        //TODO: check if spell is equipped in a utility slot, then do this
+        player.getAbilities().mayfly = !player.getAbilities().mayfly;
+        player.onUpdateAbilities();
+    }
+
+    @SubscribeEvent
+    public void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        if (event.side.isServer() && event.player.level.getDayTime() % 20 == 0 && event.phase == TickEvent.Phase.START && event.player.getAbilities().mayfly) {
```

Comment:
This needs vastly expanded, but it might be better to wait for the utility slots on the armor to be done. 
Because of how items and item registration works, we can't use the updateEquipment how I thought we could, so instead we'll have to do a check in here for whether or not the player has the items. 

This module will probably need 2 subscribe methods, though it will need some testing to see if that's true. This playerTick will need to check if the player has flight equipped and select the highest level, then use the rest to reduce cost. It will also be here that mayfly will be set depending on whether or not a telekinetic utility module is equipped. 
We also might need a damage interception subscribe method for when the player hits the ground and takes fall damage. I know that mayfly prevents fall damage from going through, but we do want to charge the mana tick cost if the player falls from too high.

I'll try to make a method that will return a list of all modules of a type in PassiveSpellItem.